Fetcher core crate

nachog00 opened the pull request:
This is a continuation of #37 .

nachog00 committed:
Initialised cargo-fetcehr crate

nachog00 committed:
initialised file structure

nachog00 committed:
new error.rs module

nachog00 committed:
new cache.rs module

nachog00 committed:
new binaries module

nachog00 committed:
new binaries module

nachog00 committed:
updated fetcher-core lib.rs

nachog00 committed:
format cargo.toml

nachog00 committed:
new get_path function in cache

nachog00 committed:
new error "InvalidResource"

nachog00 committed:
remove binary_cofig module
not nieeded anymore after next refactor

nachog00 committed:
.

nachog00 committed:
removed unnecesary implementations of ResourceType

nachog00 committed:
implemented correct Resource trait

nachog00 committed:
implemented Resource for Binaries

nachog00 committed:
updated public api to use latest Resource implementation

nachog00 committed:
added tokio as a dep of fetcher-core

nachog00 committed:
more logging

nachog00 committed:
implemented get_name as Resource for Binaries

nachog00 committed:
fix: &Cache for get_path

nachog00 committed:
removed unnecesary 'get' fn and implemented dummy fn's for Binaries's Resource trait

nachog00 committed:
updated get_result 's types

nachog00 committed:
added assert!(false) in dummy function to see outpu on fail

ala-mode committed:
remove trait

ala-mode committed:
Remove resource.rs

ala-mode committed:
cleanup

ala-mode committed:
suppress warnings

ala-mode committed:
removed mod.rs files and flattened crate

ala-mode committed:
error with .get() method

nachog00 committed:
Merge pull request #4 from ala-mode/flatten_crate

Flatten crate

nachog00 committed:
fix: some faulty imports and change modules visibility

nachog00 committed:
refactor Binaries.get into a more readable pattern

nachog00 committed:
new Binaries method to get its type name

nachog00 committed:
Moved all shasums into fetcher-core/shasums/
since these are inherent to fetcher-core and not the workspace

nachog00 committed:
New type_name attribute of Binaries
This is useful in building unique keys as well as for path to the spcific resource

nachog00 committed:
New utils.rs

nachog00 committed:
Merge pull request #37 from nachog00/fetcher-core_crate

fetcher-core crate

nachog00 committed:
Moved get_shasum into a Binaries owned method

nachog00 committed:
Merge pull request #40 from nachog00/fetcher-core_crate

minor fixes and utils

nachog00 committed:
New deps: hex and sha2

nachog00 committed:
finished method get_shasum for binaries
Also ported function to encode file into hash

nachog00 committed:
small cleanup

nachog00 committed:
fixed Binaries.get_name visibility

nachog00 committed:
removed unecessary method _get_checksum fo Binaries

nachog00 committed:
Implemented confirm function

nachog00 committed:
implemented verify method for Binaries

nachog00 committed:
Implemented get_bytes method for Binaries

nachog00 committed:
fix: return type of bytes method

nachog00 committed:
Implemented get_Version_command method and updated get_version_string

nachog00 committed:
fix: typo in shasums directory

nachog00 committed:
ignored fetched_resources to avoid checking it in by mistake

nachog00 committed:
fix imports

nachog00 committed:
fix: get_version_string type

nachog00 committed:
Fix: proper path in get_shasum method

nachog00 committed:
implemented missing version checking

nachog00 committed:
logs: changed confirm message

nachog00 committed:
Added ErrorType for MissingShasum
Probably wont be needed in the future as this should be taken into account from the source code

nachog00 committed:
Initialised fetch method with old logic

nachog00 committed:
added reqwest as dep in fetcher-core

nachog00 committed:
Added cert/ dir to fetcher-core

nachog00 committed:
fixed certificate ingest

nachog00 committed:
Finished implementing new fetch logic

nachog00 committed:
fix: correct server url

nachog00 committed:
removed all unncecesary _underscored methods

nachog00 committed:
implemented get_result function
also updated return types in .get

nachog00 committed:
removed dbg of shasum_path var

ala-mode committed:
Merge pull request #41 from nachog00/port_fetcher_logic

port fetcher logic

nachog00 committed:
removed 'I'm confirming/verifying/fetching' logs

nachog00 committed:
hash checking logs in separate lines
for better readability

nachog00 committed:
loads of new logs

nachog00 committed:
verifying log and indenting for confirming sub log

nachog00 committed:
indented verification result log

nachog00 committed:
removed whole fetcher crate

nachog00 committed:
Updated fetcher-core's package name

nachog00 committed:
Made fetcher-core a dep of services

nachog00 committed:
Reverted process invocation's in services to PATH available bins (temp)
Upcomming PR's will implement fetcher-builder crate and/or direct consumeing of fetcher-core from services

nachog00 committed:
Merge pull request #51 from nachog00/remove_fetcher_crate

remove fetcher crate

ala-mode committed:
cargo update

ala-mode commented on Cargo.lock:
```diff
@@ -5666,7 +5666,7 @@ dependencies = [
 
 [[package]]
-name = "zingo-infra-fetcher"
+name = "zingo-infra-fetcher-core"
```
clear change here and similar. thumbs up.

ala-mode commented on .gitignore:
```diff
@@ -1 +1,3 @@
 target
+fetcher-core/design-chat.md
```
this doesn't appear to be checked in, so I'd take this out.

nachog00 replied:
yeah, that's a quick local dev thing for myself that i missed

ala-mode commented on Cargo.toml:
```diff
@@ -2,8 +2,8 @@
   "testutils",
-  "fetcher"
-]
+  "fetcher-core"
```
nice to have this so clear in this PR!

ala-mode commented on fetcher-core/Cargo.toml:
```diff
@@ -1,11 +1,10 @@
-build = "src/build.rs"
 
-[build-dependencies]
+[dependencies]
```
clear change away from build deps...

nachog00 committed:
Merge pull request #52 from ala-mode/cargo_update_

cargo update

nachog00 commented:
Merging this will (temporarily) take `services` back to using PATH available binaries if none passed.
Upcoming PR's will integrate it to `fetcher-core` either through `fetcher-builder`  or directly.

ala-mode commented on fetcher-core/src/binary.rs:
```diff
@@ -0,0 +1,321 @@
+            .expect("stdout to happen")
+            .read_to_string(&mut std_out)
+            .expect("writing to buffer to complete");
+        // version
```
I see commented out code.

nachog00 replied:
Since the whole version checking thing was a discussion topic then, i left it in. If we agree on the new way, we can take it out completely.

ala-mode commented on fetcher-core/src/binary.rs:
```diff
@@ -0,0 +1,321 @@
+            .expect("command with --version argument and stddout + stderr to be created");
+
+        let mut std_out = String::new();
+        // let mut std_err = String::new();
```
Same.

nachog00 replied:
Same as above

ala-mode commented on fetcher/src/build.rs:
```diff
@@ -1,363 +0,0 @@
-    )
-    .expect("reqwest to ingest cert");
-    println!("cert ingested : {:?}", cert);
-
```
and removed here, with this commit: `635f4c5 removed whole fetcher crate`

nachog00 committed:
implemented zainod_exists test

nachog00 committed:
implemented zainod_exists test

ala-mode reviewed (commented):
following the thread back leads [here](https://github.com/zingolabs/infrastructure/issues/31) for overall info.

I love that I can see the cert and hashsums moved without changes.

checking out `commit 2e1b1522fce65277c6c2fabf7ef588f220bea52e (HEAD -> dev, zingo/dev)`, then running git diff fdb7dca shows
for example, line 311 : `// let s_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(9, 9, 9, 9)), 9073);` which is then reintroduced to the code in the current PR, see comments for more details.

ala-mode commented:
I am getting `thread 'launch_localnet_zainod_zebrad' panicked at services/src/indexer.rs:143:42:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
` and similar on `cargo nextest run` attempts.

Seems like it's not finding binaries. Is there a special invocation to get fetcher to run now? If so, where is that documented (messages here on github, or in a README)? Or am I doing it wrong?

nachog00 committed:
New test_utils module in fetcher-core

nachog00 committed:
New test_utils module in fetcher-core

nachog00 committed:
used new module on tests

nachog00 committed:
used new module on tests

nachog00 commented:
> I am getting `thread 'launch_localnet_zainod_zebrad' panicked at services/src/indexer.rs:143:42: called `Result::unwrap()`on an`Err`value: Os { code: 2, kind: NotFound, message: "No such file or directory" }` and similar on `cargo nextest run` attempts.
> 
> Seems like it's not finding binaries. Is there a special invocation to get fetcher to run now? If so, where is that documented (messages here on github, or in a README)? Or am I doing it wrong?



> Merging this will (temporarily) take `services` back to using PATH available binaries if none passed. Upcoming PR's will integrate it to `fetcher-core` either through `fetcher-builder` or directly.

TO be clear, this means: `services` will look for binaries on the global PATH, and it s passing tests as of now that way.

nachog00 commented:
Turned this back to DRAFT as we agreed on keep building here until `services` actually consumes `fetcher-core`

nachog00 committed:
indenting on fethc method logs

nachog00 committed:
moved tests away from lib.rs into tests/

nachog00 committed:
moved tests away from lib.rs into tests/

nachog00 committed:
Removed commented out code

nachog00 committed:
removed tests dir, not relevant to this PR

nachog00 committed:
Completed Binaries enum

completed version strings and commands

completed bytes

completed names

nachog00 committed:
removed unecessary line from gitignore

ala-mode committed:
Merge pull request #63 from nachog00/add_missing_binaries

Add missing binaries

ala-mode committed:
Merge pull request #62 from nachog00/cleanup_commented_code

Removed commented out code

nachog00 committed:
Merge branch 'fetcher-core_crate' into testing_fetcher-core

nachog00 committed:
added 'exists' tests for all recently added binaries

nachog00 committed:
added 'exists' tests for all recently added binaries

nachog00 committed:
Fix: get_path method was returning an error if the file didnt exist yet
This didnt make sense as this method was needed to even get the path to write the not yet existing file.

nachog00 committed:
Docuemnted Cache struct

docs for 'new' method

docs for 'load' method

docs for 'store' method

docs for 'exists' method

docs for 'get_path' method

nachog00 committed:
replaced repeated code with method

nachog00 committed:
removed unncessary method

nachog00 committed:
Documented lib.rs (fetcher-core)
also removed unecessary enum

nachog00 committed:
Documented binary.rs (fetcher-core)

nachog00 committed:
Merge pull request #64 from nachog00/fix_get_path_bug

Fix get_path method bug

nachog00 committed:
Try out integrating zebrad in services using fetcher-core

nachog00 committed:
Shasum files now imported at compile time
This avoids location issues when consumed

nachog00 committed:
Cert file included at compile time

nachog00 committed:
Removed utils.rs completely as neither helper function was being used anymore

nachog00 committed:
removed old comment

nachog00 committed:
fix: removed 'mod utils' import

nachog00 committed:
implemented fetcher-core's zcashd in services' Validator

ala-mode committed:
Merge pull request #66 from nachog00/documenting

docs for fetcher-core (commit into feature branch)

ala-mode committed:
Merge pull request #67 from nachog00/include_bytes_x_manifest_dir

Include byes at compile time (into feature branch)

nachog00 committed:
moved binary_exists into unit_tests.rs

ala-mode committed:
Merge pull request #61 from nachog00/testing_fetcher-core

testing fetcher core (into feature branch)

ala-mode committed:
Merge pull request #65 from nachog00/cache_struct_cleanup

LGTM! cache struct cleanup (into feature branch)

nachog00 committed:
Merge branch 'fetcher-core_crate' into services_integration_direct

nachog00 committed:
changed signature for Idexer.launch into an async fn

nachog00 committed:
Indexer implementation of Zainod and Lwd now async and using fetcher-core's resources

nachog00 committed:
Adapted all usages of now async launch methods to use .await now

nachog00 committed:
fix typo

ala-mode committed:
Merge pull request #68 from nachog00/logs_cleanup

logs for Binaries get (into feature branch)

ala-mode committed:
Merge pull request #69 from nachog00/services_integration_direct

services integration direct

because it's so integral to #39 and that one is already monolithic, I'm going to put everything in one place.

nachog00 committed:
Revert "Merge pull request #69 from nachog00/services_integration_direct"

This reverts commit ef7b9c85b6790402b11a5ee9630a71e5d6a00f45, reversing
changes made to e3e7b23a1ad11a4a11f56e314941cdc8bdd2e390.

ala-mode committed:
Merge pull request #73 from nachog00/fetcher-core_crate

Revert "Merge pull request #69 from nachog00/services_integration_direct"

nachog00 committed:
Merge branch 'dev' into fetcher-core_crate

nachog00 committed:
Merge branch 'dev' into fetcher-core_crate

ala-mode committed:
Merge pull request #74 from nachog00/fetcher-core_crate

Fix merge conflicts with latest dev

(merging into feature branch)

ala-mode commented:
Summary [ 153.837s] 46 tests run: 46 passed (6 slow), 5 skipped
